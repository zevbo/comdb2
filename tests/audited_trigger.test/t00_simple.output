cdb2_open rc -1 cdb2_get_dbhosts: no default_type entry in comdb2db config.
cdb2_open rc -1 cdb2_get_dbhosts: no default_type entry in comdb2db config.
cdb2_open rc -1 cdb2_get_dbhosts: no default_type entry in comdb2db config.
cdb2_open rc -1 cdb2_get_dbhosts: no default_type entry in comdb2db config.
cdb2_open rc -1 cdb2_get_dbhosts: no default_type entry in comdb2db config.
cdb2_open rc -1 cdb2_get_dbhosts: no default_type entry in comdb2db config.
cdb2_open rc -1 cdb2_get_dbhosts: no default_type entry in comdb2db config.
cdb2_open rc -1 cdb2_get_dbhosts: no default_type entry in comdb2db config.
cdb2_open rc -1 cdb2_get_dbhosts: no default_type entry in comdb2db config.
cdb2_open rc -1 cdb2_get_dbhosts: no default_type entry in comdb2db config.
cdb2_open rc -1 cdb2_get_dbhosts: no default_type entry in comdb2db config.
cdb2_open rc -1 cdb2_get_dbhosts: no default_type entry in comdb2db config.
cdb2_open rc -1 cdb2_get_dbhosts: no default_type entry in comdb2db config.
