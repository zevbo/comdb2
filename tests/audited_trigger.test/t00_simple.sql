CREATE TABLE t(num int)$$

CREATE LUA TRIGGER t_audit ON (TABLE t FOR INSERT) AUDITED

SELECT "------table and procedure creation------"
SELECT * FROM comdb2_tables ORDER BY tablename

INSERT INTO t VALUES(5)
INSERT INTO t VALUES(10)
DELETE FROM t WHERE num=10

SELECT sleep(1)

SELECT "------table filling------"
SELECT * from "$audit_t" ORDER BY new_num

DROP TABLE t 
DROP LUA TRIGGER t_audit
DROP TABLE "$audit_t"