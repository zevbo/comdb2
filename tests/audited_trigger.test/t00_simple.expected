[CREATE TABLE t(num int)] rc 0
[CREATE LUA AUDIT TRIGGER t_audit ON (TABLE t FOR INSERT)] rc 0
("------table and procedure creation------"='------table and procedure creation------')
[SELECT "------table and procedure creation------"] rc 0
(tablename='$audit_t')
(tablename='sqlite_stat1')
(tablename='sqlite_stat4')
(tablename='t')
[SELECT * FROM comdb2_tables ORDER BY tablename] rc 0
(rows inserted=1)
[INSERT INTO t VALUES(5)] rc 0
(rows inserted=1)
[INSERT INTO t VALUES(10)] rc 0
(rows deleted=1)
[DELETE FROM t WHERE num=10] rc 0
(sleep(1)=1)
[SELECT sleep(1)] rc 0
("------table filling------"='------table filling------')
[SELECT "------table filling------"] rc 0
(type='add', tbl='t', old_num=NULL, new_num=5)
(type='add', tbl='t', old_num=NULL, new_num=10)
[SELECT type, tbl, old_num, new_num from "$audit_t" ORDER BY new_num] rc 0
[DROP TABLE t] rc 0
[DROP LUA TRIGGER t_audit] rc 0
[DROP TABLE "$audit_t"] rc 0
